h/static/scripts/directive/annotation.js

```diff
@@ -781,6 +781,49 @@ function AnnotationController(
   init();
 }
 
+function link(scope, elem, attrs, controllers) {
+  var ctrl = controllers[0];
+  var thread = controllers[1];
+  var threadFilter = controllers[2];
+  var counter = controllers[3];
+
+  elem.on('keydown', ctrl.onKeydown);
+
+  // FIXME: Replace this counting code with something more sane.
+  if (counter !== null) {
```

Comment:
By the way it looks to me like this guard (and the one below it as well) is probably only hear because the required controllers are `null` when running the tests!